storage-proofs-update/src/circuit_poseidon.rs

```diff
@@ -0,0 +1,667 @@
+    // old/un-updated data (here called TreeROld).
+    pub root_r_old: Option<Fr>,
+    // Root of the replica tree build over the new/updated data's replica (TreeRNew).
+    pub root_r_new: Option<Fr>,
+    // Generate three Merkle proofs (TreeROld, TreeDNew, TreeRNew) for each of this partition's
+    // challenges.
+    pub challenge_proofs: Vec<ChallengeProof<TreeR>>,
+}
+
+impl<TreeR> PrivateInputs<TreeR>
+where
+    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+{
+    pub fn new(comm_c: TreeRDomain, challenge_proofs: &[ChallengeProofVanilla<TreeR>]) -> Self {
+        let root_r_old: Fr = challenge_proofs[0].proof_r_old.root().into();
+        let root_r_new: Fr = challenge_proofs[0].proof_r_new.root().into();
+
+        let challenge_proofs: Vec<ChallengeProof<TreeR>> = challenge_proofs
+            .iter()
+            .cloned()
+            .map(ChallengeProof::from)
+            .collect();
+
+        PrivateInputs {
+            comm_c: Some(comm_c.into()),
+            root_r_old: Some(root_r_old),
+            root_r_new: Some(root_r_new),
+            challenge_proofs,
+        }
+    }
+
+    pub fn empty(sector_nodes: usize) -> Self {
+        let challenge_count = challenge_count(sector_nodes) * partition_count(sector_nodes);
+        PrivateInputs {
+            comm_c: None,
+            root_r_old: None,
+            root_r_new: None,
+            challenge_proofs: vec![ChallengeProof::empty(sector_nodes); challenge_count],
+        }
+    }
+}
+
+pub struct EmptySectorUpdateCircuit<TreeR>
+where
+    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+{
+    pub pub_params: PublicParams,
+    pub pub_inputs: PublicInputs,
+    pub priv_inputs: PrivateInputs<TreeR>,
+}
+
+impl<TreeR> CircuitComponent for EmptySectorUpdateCircuit<TreeR>
+where
+    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+{
+    type ComponentPrivateInputs = ();
+}
+
+impl<TreeR> EmptySectorUpdateCircuit<TreeR>
+where
+    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+{
+    pub fn blank(pub_params: PublicParams) -> Self {
+        let pub_inputs = PublicInputs::empty();
+        let priv_inputs = PrivateInputs::<TreeR>::empty(pub_params.sector_nodes);
+        EmptySectorUpdateCircuit {
+            pub_params,
+            pub_inputs,
+            priv_inputs,
+        }
+    }
+}
+
+impl<TreeR> Circuit<Fr> for EmptySectorUpdateCircuit<TreeR>
+where
+    TreeR: MerkleTreeTrait<Hasher = TreeRHasher>,
+{
+    /// This circuit is NOT AUDITED, USE AT YOUR OWN RISK.
+    fn synthesize<CS: ConstraintSystem<Fr>>(self, cs: &mut CS) -> Result<(), SynthesisError> {
+        #[allow(unused_variables)]
+        let EmptySectorUpdateCircuit {
+            pub_params:
+                PublicParams {
+                    sector_nodes,
+                    challenge_count,
+                    challenge_bit_len,
+                    partition_count,
+                    partition_bit_len,
+                    apex_leaf_count,
+                    apex_leaf_bit_len,
```

Comment:
I tried ignoring them but I couldn't figure it out for some reason, re PublicParams see above, I wanted to keep the same PublicParams, which could be modified to better suite both circuit but I also wanted to avoid affecting SHA circuit in any way right now.

Reply:
fixed